NetTally/ReorderTasksWindow.xaml.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using NetTally.Utility;
+using NetTally.ViewModels;
+
+namespace NetTally
+{
+    /// <summary>
+    /// Interaction logic for ReorderTasks.xaml
+    /// </summary>
+    public partial class ReorderTasksWindow : Window, INotifyPropertyChanged
+    {
+        #region Constructor and variables
+        public ListCollectionView TaskView { get; }
+
+        MainViewModel MainViewModel { get; }
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        public ReorderTasksWindow()
+        {
+            InitializeComponent();
+        }
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mainViewModel">The primary view model of the program</param>
+        public ReorderTasksWindow(MainViewModel mainViewModel)
+        {
+            InitializeComponent();
+
+            MainViewModel = mainViewModel;
+
+            MainViewModel.PropertyChanged += MainViewModel_PropertyChanged;
+
+            // Create filtered, sortable views into the collection for display in the window.
+            if (MainViewModel.TaskList != null)
+            {
+                TaskView = new ListCollectionView(MainViewModel.TaskList);
+            }
+
+            // Initialize starting selected positions
+            TaskView.MoveCurrentToPosition(-1);
+
+            // Set the data context for binding.
+            DataContext = this;
+
+            // Update the displayed order immediately
+            UpdateTaskList();
+        }
+        #endregion
+
+        private void UpdateTaskList()
+        {
+            VoteCounter.Instance.OrderedTaskList = TaskView.SourceCollection as List<string>;
+            OnPropertyChanged("Votes");
```

Comment:
So, while I wrote all that out for the general case, there are a number of caveats on what all needs to be changed that's a bit too complicated to write in.  I've gone through and improved the handling of several aspects of the collections and events tied to them in different parts of the code.

I'll most likely end up just taking the current pull request (as I have, manually) and just add all the corrections on top, rather than ask for a new PR entirely.  It's going to be a pain to move this (and other VS2015 changes) over to the VS2017 branch regardless, so I might as well work on that as well.